src/compiler/Compiler.js

```diff
@@ -92,7 +92,13 @@ module.exports = class Compiler extends EventEmitter {
    * Create fork process and attach necessary event listeners.
    */
   initFork({ platform, options }: { platform: Platform, options: * }) {
-    const fork = new Fork({ platform, options });
+    let fork;
+    try {
+      fork = new Fork({ platform, options });
+    } catch (err) {
+      this.emit(Events.BUILD_FAILED, { platform, message: err.toString() });
+      return null;
```

Comment:
Looks like returning `null` here will break code at line 43?

![image](https://user-images.githubusercontent.com/1174278/40418520-e4788520-5e82-11e8-96d4-3944512fec24.png)

It checks the `.isProcessing` property. Maybe it's better to move emitting event to the callsite (line 37)?

Reply:
Agree. This going to crash the whole thing. With nice error and ⚠️icon next to the platform.